### ISSUE TITLE
Consider making dependency on systemd optional

### ISSUE THREAD

**Issue by tbzatek** [sid-project/sid-mvp#17]
One hard requirement for even considering usage of `sid` within `udisks` is support for non-`systemd` Linux distributions. Systems running either forks like `eudev` or ripped-out and heavily patched `systemd-udev` from the `systemd` tree (nasty, but necessary).

Technically certain `systemd` API got reimplemented by independent projects, e.g. `elogind`, however most of the `systemd` core init daemon API is not available.

We simply can't afford limiting `udisks` availability to certain init systems, that would be too big regression given the current widespread of `udisks` among desktop and embedded systems. Maintaining two code paths keeping `udev` supported makes no sense either as the daemon is largely async and needs to be fine-tuned to prevent race conditions, etc. I cannot imagine continuous testing and support for both options.

Cc: @vojtechtrefny, @vpodzime

**Comment by tbzatek** [comment]
Looking at the sources, seems like the majority of `libsystemd` usage is about `sd-event`. And looking at `systemd` sources, the whole `sd-event` part looks reasonably independent to the rest of the code, meaning it could be ripped-out and possibly made a compat library (by the distributors).

_(I personally wonder why it's all kept in a monolithic library while such general stuff could easily be distributed separately)_

**Comment by prajnoha** [comment]
Yes, ``sid`` uses two parts from ``systemd`` (I'm also including ``udev`` here):

  - event loop functionality (``sd-event``)
  - udev monitoring functionality (``libudev``)

I tried to count with this in a way with the ``sd-event`` usage in ``sid`` -  this part is wrapped in one place which can be easily replaced with a different implementation of event loop. The event loop in ``sd-event`` is pretty generic which does not differ much from other event loop library implementations.

**Comment by tbzatek** [comment]
At the end there wouldn't be need for any rewrite since `sd-event` is apparently used and exposed within `libelogind`. Nowadays `elogind` is pretty widespread on non-systemd systems and is basically a fork and subset of `systemd` sources.

Quote from the https://github.com/elogind/elogind repo docs:
> Elogind also provides "libelogind", which is a subset of the facilities offered by "libsystemd". 

So the only thing needed would probably be some autotools magic and different includes, similar to this `udisks` commit: https://github.com/storaged-project/udisks/commit/884b63045c6690cc9e7426c4318cbe3950447507

Let me play a little bit with that and see if it works. No action needed from your side for the moment.

**Comment by prajnoha** [comment]
Well, yes, and systemd is now quite widespread too, also udev is part of systemd these days. But there's still the possibility to support different event loop mechanism. I'm not bound firmly to any concrete implementation - it's just I'm using libudev monitoring which is part of libudev which is part of udev which in turn is part of systemd these days so it comes natural for me to use libsystemd's event loop by default :)
